File: tks/ApplicationCore/DomainModel/src/main/kotlin/domain/model/values/Email.java
package domain.model.values;

import domain.exceptions.TypeValidationFailedException;
import lombok.Value;
import org.apache.commons.validator.routines.EmailValidator;

@Value
public class Email {
    String value;

    public Email(String value) throws TypeValidationFailedException {
        if(!EmailValidator.getInstance(false).isValid(value)) throw new TypeValidationFailedException();
        this.value = value;
    }
}

File: tks/ApplicationCore/DomainModel/src/main/kotlin/domain/model/values/Email.kt
package domain.model.values

import domain.exceptions.TypeValidationFailedException
import lombok.Value
import org.apache.commons.validator.routines.EmailValidator

@JvmInline
value class Email(val value: String) {
    init {
        if (!EmailValidator.getInstance(false).isValid(value)) throw TypeValidationFailedException()
    }
}